src/main/kotlin/net/ccbluex/liquidbounce/utils/combat/TargetTracker.kt

```diff
@@ -84,49 +84,55 @@ open class TargetSelector(
         this(defaultPriority, DummyRangedValueProvider(range))
 
     var closestSquaredEnemyDistance: Double = 0.0
+        private set
 
     private val range = rangeValue.register(this)
     private val fov by float("FOV", 180f, 0f..180f)
     private val hurtTime by int("HurtTime", 10, 0..10)
     private val priority by enumChoice("Priority", defaultPriority)
 
+    private val targetComparator: Comparator<LivingEntity> = compareBy<LivingEntity> {
+        when (priority) {
+            // Lowest health first
+            TargetPriority.HEALTH -> it.getActualHealth()
+            // Closest to your crosshair first
+            TargetPriority.DIRECTION -> RotationUtil.crosshairAngleToEntity(it)
+            // Oldest entity first
+            TargetPriority.AGE -> -it.age
+            // With the lowest hurt time first
+            TargetPriority.HURT_TIME -> it.hurtTime
+            // Closest to you first
+            else -> 0 // Do nothing because we will compare by distance later
+        }
+    }.thenComparingInt { entity ->
+        // Sort by entity type (player first, then hostile, then other)
+        when (entity) {
+            is PlayerEntity -> 0
+            is HostileEntity -> 1
+            else -> 2
+        }
+    }.thenComparingDouble {
+        // Sort by distance (closest first) - in case of tie at priority level
+        it.squaredBoxedDistanceTo(player.eyePos)
+    }
```

Comment:
This is wrong. The goal is to sort by distance first, then sort by priority and as last sort by entity type.

Reply:
That is still wrong lol